### ISSUE TITLE
Refactor instant delivery provider to return new object structure 

### ISSUE THREAD

**Issue by mirgj** [aws-samples/aws-last-mile-delivery-hyperlocal#51]
Refactor instant delivery provider to return orders as list of segments that can also have mixed pick-up/drop-off. The same structure will be used for same day delivery

```
{
  jobId: '',
  driverId: '', // only for instant delivery
  driverIdentity: '', // only for instant delivery
  route: {
    distance: {
      value: 1152,
      unit: 'm', // meter
    },
    time: {
      value: 1160,
      unit: 'sec' //seconds
    },
    pointsEncoded: 'cxl_cBqwvnS|Dy@ogFyxmAf`IsnA|CjFzCsHluD_k@hi@ljL', //as per path polyline
  },
  segments: [
    {
      orderId: '',
      index: 1,
      from: { lat: 1, long: 2 },
      to: { lat: 1, long: 2 },
      segmentType: 'TO_ORIGIN/TO_DESTINATION',
      route: {
        distance: {
          value: 1152,
          unit: 'm', // meter
        },
        time: {
          value: 1160,
          unit: 'sec' //seconds
        },
        pointsEncoded: 'cxl_cBqwvnS|Dy@ogFyxmAf`IsnA|CjFzCsHluD_k@hi@ljL', //as per path polyline
      },
    },
    {
      orderId: '',
      index: 2,
      from: { lat: 1, long: 2 },
      to: { lat: 1, long: 2 },
      segmentType: 'TO_ORIGIN/TO_DESTINATION',
      route: {
        distance: {
          value: 478,
          unit: 'm', // meter
        },
        time: {
          value: 68,
          unit: 'sec' //seconds
        },
        pointsEncoded: '_p~iF~ps|U_ulLnnqC_mqNvxq`@, //as per path polyline
      },
    },
  ],
}
```

**Comment by sperka** [comment]
closed by #107

**Comment by mirgj** [comment]
reopening since this has to track also the backend
